libs/frontend/feature-render/src/lib/render.ts

```diff
@@ -28,12 +28,12 @@ export const computePropsContent = (
 
   const content: IContent = component.children?.length
     ? component.children.map((child) => renderComponent(site, child, renderMode))
-    : parseI18n(site, component.content)
+    : parseI18n(site, data.content)
 
   const props = {
     ...data.attrs,
     ...events.native,
-    class: data.mixins.concat(component.id),
+    class: data.mixins.concat(component.id, component.reusableSourceId ?? []),
```

Comment:
Rather than merge the styles (like how I tried to do it with component inheritance before), I'm just including the reusable component ID in the class list. I wonder if style precedence is always going to work 🤔

Reply:
Yes, I think so!
From [this sandbox](https://stackblitz.com/edit/stackblitz-starters-vupjg7?file=index.html) it seems that the order of classes on an element will not effect the output (so `class="r g b"` and `class="g b r"` will lead to the same result). Instead, it is the order of styles defined in `<style>` that will effect the output, assuming that they all have the same priority (weight/specificity).

In `pubstudio-builder/libs/frontend/ui-build/src/components/BuildContent.vue:10` we render `<ReusableStyle />` first, then `<ComponentStyle />`, so component custom styles will always override reusable styles, unless the priority of styles defined in `<ReusableStyle />` is higher than that defined in `<ComponentStyle />`. It seems to work fine, at least for now. 😅

Reply:
Right but I think we will run into trouble since the reusable component style will override the instance's style if it is rendered later (they're both in `ComponentStyle`). I think there are some straightforward ways to solve the problem though:
1. Since instances will always be created after the reusable component, and IDs are sequential, sorting them before render will ensure the instance style always overrides the reusable style. This seems a little hacky though.
2. Switch back to keeping reusable components separate, and render `ReusableComponentStyle` before `ComponentStyle`.
3. Keep components in the page, but change their IDs so they are more easily sorted/filtered. E.g. when a component is converted to reusable, change the ID from `test-c-1` to `test-rc-1` or similar.

I don't think it's necessary to solve this problem right away though, it's better to get everything else working first!